api.c

```diff
@@ -503,10 +503,20 @@ int tcmu_emulate_evpd_inquiry(
 						   ASC_INVALID_FIELD_IN_CDB, NULL);
 		}
 
-		max_xfer_length = tcmu_get_attribute(dev, "hw_max_sectors");
-		if (max_xfer_length < 0) {
-			return tcmu_set_sense_data(sense, ILLEGAL_REQUEST,
-						   ASC_INVALID_FIELD_IN_CDB, NULL);
+		/*
+		 * Daemons like runner may override the user requested
+		 * value due to device specific limits.
+		 */
+		if (dev->max_xfer_len) {
+			max_xfer_length = dev->max_xfer_len / block_size;
+		} else {
+			max_xfer_length = tcmu_get_attribute(dev,
+							     "hw_max_sectors");
+			if (max_xfer_length < 0) {
+				return tcmu_set_sense_data(sense,
+						ILLEGAL_REQUEST,
+						ASC_INVALID_FIELD_IN_CDB, NULL);
```

Comment:
Shouldn't this case the sense return:  HARDWARE_ERROR and ASC_INTERNAL_TARGET_FAILURE ?